### DISCUSSION TITLE
preflight - kicad 6 DRC exists for zones,  KiCAD 5 not

### DISCUSSION THREAD

**Discussion by mdeweerd** [INTI-CMNB/KiBot#294] [General]
Just got a board back that has VDD and GND shorted :-( .

I probably forgot to explicitly refill the zones after the last changes, but it also turns out that KiCAD 5 DRC does not detect the short - KiCAD 6 does.

Here is it:
![image](https://user-images.githubusercontent.com/1504752/190499668-352ab139-78bb-4895-b921-320e5d4f83b7.png)

Further I did not have the preflight drc/erc/zone fill options set for this project.

For "educational purposes", I enabled DRC and ERC in preflight, adjusted the filter expressions and regenerated my project in diff mode, using the PCB difference functionnality to compare the "new" output with the previous one.

With `  check_zone_fills: false` : the short is not detected and makes it to the gerbers for production.  There is no difference in the PCBs.

With `  check_zone_fills: true` : The difference output still shows no difference, but the short does not make it to the gerbers.
So this means I can't trust the difference output.

This is the fixed zone fill.
![image](https://user-images.githubusercontent.com/1504752/190501509-839b5070-0db6-4789-b0e4-0816bc9f1622.png)

So in summary:
- I am going to move to KiCAD 6.  It has clear advantages - it's better with zone fill DRCs and it's also better at detecting issues with footprints - and I guess much more to discover.
- The difference functionnality also needs to use the pre-flight options to ensure that the difference outputs correspond to the actual manufacturing file differences.
- The manufacturer's DRC is not that great - they did not detect the violations either (clearance was not good everywhere), so I need to continue to check my files in mutliple ways.
